lib/fiware-iotagent-lib.js

```diff
@@ -239,6 +224,75 @@ function deactivate(callback) {
     contextServer.stop(callback);
 }
 
+/**
+ * Start the HTTP server either in single-thread or multi-thread (multi-core).
+ *
+ * @param {Object} newConfig            Configuration of the IoT Agent.
+ * @param {Object} iotAgent             The IoT Agent Objects.
+ * @param {Function} callback           The callback function.
+ */
+function startServer(newConfig, iotAgent, callback) {
```

Comment:
With this modification is supposed that the IOTA using the library to delegate the start of the server in the library. Up to know, as far as I remember, the IOTA code (not the library) is responsible of starting the server.

Is my understanding correct?

Reply:
Yes and no. The IoT Library is responsible to configure the multi-threading or single-threading but at the end, the function is calling the IOTA (iotAgent) through the call to iotAgent.start(newConfig, callback).

Reply:
Ok, I see... the IOTA code passes a its own function in the call to startServer() now in the library.

NTC